src/transformers/cache_utils.py

```diff
@@ -448,3 +448,109 @@ def reset(self):
             # In-place ops prevent breaking the static address
             self.key_cache[layer_idx].zero_()
             self.value_cache[layer_idx].zero_()
+
+class WhisperStaticCache(Cache):
```

Comment:
Thanks for this great first start @huseinzol05! With @gante, we were discussing how the design of the static k/v cache should look for encoder-decoder models, and we distilled the design options down to two possibilities:
1. Hold a tuple of `StaticCache` caches, e.g. as proposed [here](https://github.com/huggingface/transformers/pull/28931/files#r1483312510)
2. Add a new Cache classes specific to encoder-decoder models, e.g. those with the attributes:
    - `key_cache` (same as decoder-only self-attn)
    - `value_cache` (same as decoder-only self-attn)
    - `cross_key_cache` (new for enc-dec cross-attn)
    - `cross_value_cache` (new for enc-dec cross-attn)

Option 1 doesn't require any new Cache classes, so should be easier to maintain! Thus, we were thinking this would be the best design option for Whisper (and other encoder-decoder models in the library, such as BART). Would be curious to hear you opinions here, having had a go at option 2

Reply:
@huseinzol05 this is great work!

I'm heavily biased towards option 1, especially now that we are seeing more cache types. For instance, we could easily plug in the quantized cache as the decoder cache with 0 code overhead, if we design Whisper to support a tuple of `Cache` objects through `past_key_values` 🤗

Reply:
im good with anything